modules/shared/src/main/scala/org/tessellation/ext/derevo/snakeEncoder.scala

```diff
@@ -0,0 +1,18 @@
+package org.tessellation.ext.derevo
+
+import derevo.{Derivation, NewTypeDerivation}
+import io.circe.Encoder
+import io.circe.magnolia.configured.Configuration
+import io.circe.magnolia.configured.encoder.semiauto
+import magnolia.{CaseClass, Magnolia, SealedTrait}
+
+object snakeEncoder extends Derivation[Encoder] with NewTypeDerivation[Encoder] {
+  type Typeclass[T] = Encoder[T]
+
+  def combine[T](caseClass: CaseClass[Typeclass, T]): Typeclass[T] =
+    semiauto.combine[T](caseClass)(Configuration.default.withSnakeCaseMemberNames)
+  def dispatch[T](sealedTrait: SealedTrait[Typeclass, T]): Typeclass[T] =
+    semiauto.dispatch[T](sealedTrait)(Configuration.default.withSnakeCaseMemberNames)
+
+  def instance[T]: Typeclass[T] = macro Magnolia.gen[T]
+}
```

Comment:
Wouldn't it be better to create only a configuration and then import it? I'm not sure if this works, but it looks simpler.

```scala
object jsonConfig {
 implicit val configuration:Configuration = Configuration.default.withSnakeCaseMemberNames
}
```

```scala
import derevo.derive
import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}

import org.tessellation.rosetta.jsonConfig

@derive(customizableEncoder, customizableDecoder)
sealed trait SealedTrait
case class Foo(value: String)
```

Reply:
It works but then you have to import the configuration everywhere and wrap case classes into objects. I did the same what customizableEncoder/Decoder does - just predefined the configuration and delegate encoder/decoder to magnolia.

Reply:
Oh, I got your point with importing the configuration. Indeed it's simpler. I'll change that!